cantor-base/src/main/java/com/salesforce/cantor/Events.java

```diff
@@ -261,7 +319,125 @@ default void store(String namespace,
+                              long startTimestampMillis,
+                              long endTimestampMillis) throws IOException {
+        return last(namespace, startTimestampMillis, endTimestampMillis, null, null, false);
+    }
+
+    /**
+     * Get the last (latest) event in the given namespace, with timestamp between the start and end, and metadata and dimensions
+     * matching the given queries.
+     *
+     * @param namespace the namespace identifier
+     * @param startTimestampMillis start UTC timestamp in milli-seconds
+     * @param endTimestampMillis end UTC timestamp in milli-seconds
+     * @param metadataQuery map of string to string representing a query to run against events metadata
+     * @param dimensionsQuery map of string to string representing a query to run against events dimensions
+     * @return the earliest event in the namespace with timestamp between start/end and metadata/dimensions matching the query
+     * @throws IOException exception thrown from the underlying storage implementation
+     */
+    default Event last(String namespace,
+                              long startTimestampMillis,
+                              long endTimestampMillis,
+                              Map<String, String> metadataQuery,
+                              Map<String, String> dimensionsQuery) throws IOException {
+        return last(namespace, startTimestampMillis, endTimestampMillis, metadataQuery, dimensionsQuery, false);
+    }
+
+    /**
+     * Get the last (latest) event in the given namespace, with timestamp between the start and end, and metadata and dimensions
+     * matching the given queries.
+     *
+     * @param namespace the namespace identifier
+     * @param startTimestampMillis start UTC timestamp in milli-seconds
+     * @param endTimestampMillis end UTC timestamp in milli-seconds
+     * @param metadataQuery map of string to string representing a query to run against events metadata
+     * @param dimensionsQuery map of string to string representing a query to run against events dimensions
+     * @param includePayloads flag to include payloads in the response or not; if false, event.getPayload() returns null
+     * @return the earliest event in the namespace with timestamp between start/end and metadata/dimensions matching the query
+     * @throws IOException exception thrown from the underlying storage implementation
+     */
+    default Event last(String namespace,
+                       long startTimestampMillis,
+                       long endTimestampMillis,
+                       Map<String, String> metadataQuery,
+                       Map<String, String> dimensionsQuery,
+                       boolean includePayloads) throws IOException {
+        final List<Event> events = get(namespace, startTimestampMillis, endTimestampMillis, metadataQuery, dimensionsQuery, includePayloads, false, 1);
+        return !events.isEmpty() ? events.get(0) : null;
```

Comment:
Should we check null too for events? No current impl returns null, but should the default protect against that?

Reply:
I can do that just to be safe, but since get returns a collection, it is supposed to never return null.